### PR TITLE
Add HarfTeX index field

### PR DESCRIPTION
**Pull request by zauguin**
See #66.

### CODE REVIEW

**Comment by u-fischer** [comment]
Ah. Thanks. Do you have an idea about a meaningful testfile here (for harftex and luatex)? I think the only ttc in texlive is asana.ttc.

File: src/luaotfload-auxiliary.lua [hunk 1]
```diff
@@ -105,6 +105,14 @@ luaotfload_callbacks [#luaotfload_callbacks + 1] = {
   "patch_font", set_sscale_dimens, "set_sscale_dimens",
 }
 
+-- HarfteX doesn't like 1-based indexing, so we adjust it. This doesn't hurt in
+-- LuaTeX either because LuaTeX doesn't use a `index` field in the first place.
+luaotfload_callbacks [#luaotfload_callbacks + 1] = {
+  "patch_font", function(fontdata)
+    fontdata.index = (fontdata.specification.sub or 1) - 1
```
Comment (khaledhosny) [inline src/luaotfload-auxiliary.lua]: I can make the Lua API 1-based if this looks better, but the C code is 0-based.

Comment (zauguin) [reply src/luaotfload-auxiliary.lua]: @khaledhosny Personally I think 1-based would look more natural in Lua. (Subfonts are kind of like an array of fonts, and arrays are 1-based in Lua) But as long as LuaTeX doesn't start using the same field as a 1-based index, I don't really care.

**Comment by zauguin** [comment]
@u-fischer I don't. Is the testing of the resulting PDF-file working already? Right now the only thing I can think of which can actually by seen as an effect of this setting in the logfile are missing glyphs.

**Comment by u-fischer** [comment]
@zauguin testing of pdf files works. But during normalization binary stuff like fonts are removed, so the question is if enough stays to be meaningful, but we can try. A test with missing glyphs would be imho also useful - the main point is to have at least a bit of a chance to get noticed if the engines changes something ... (I will wait with this pull request until the zero-one-stuff is decided ;-))

File: src/luaotfload-auxiliary.lua (hunk 1 above)
Comment (khaledhosny) [reply src/luaotfload-auxiliary.lua]: The font `characters` table seems to be 0-based, and that one have stronger reasons to be 1-based, so I think I'm sticking to 0 for now.

Comment (zauguin) [reply src/luaotfload-auxiliary.lua]: Actually the `characters` table doesn't behave like an array, especially the indices are not continuous. It is a associative directory indexed by Unicode Codepoints and Unicode codepoints. (or some other encoding) These Encodings are 0-based, so I actually can't think of any reason why characters should be 0-based. Anyway, harftex has harf and not lua in it's name, so conforming more to HarfBuzz than to LuaTeX conventions probably makes sense.

**Comment by u-fischer** [comment]
@khaledhosny  @zauguin  In view of https://github.com/lualatex/luaotfload/issues/392: how do the numbers now relate to the numbers output by luaotfload-tool? Is there some test that one can use here to check that everything is in sync?

**Comment by khaledhosny** [comment]
The numbers match the XeTeX/HarfBuzz APIs, I consider what luaotfload(-tool) doing to be just confusing since it takes 0-based indexes in the `font(N)` syntax (the only syntax it supports for specifying face index, but writes out 1-based indices.

**Pull request closed without merging**